packages/core/src/transports/initialize.ts

```diff
@@ -1,20 +1,52 @@
+import type { ExceptionEvent } from '../api';
 import type { Config } from '../config';
-import type { Transport, Transports } from './types';
+import type { Patterns } from '../config/types';
+import { BeforeSendHook, Transport, TransportItem, TransportItemType, Transports } from './types';
 
 export function initializeTransports(config: Config): Transports {
-  const value: Transport[] = [...config.transports];
+  const transports: Transport[] = [...config.transports];
+
+  const beforeSendHooks: BeforeSendHook[] = [];
+  if (config.beforeSend) {
+    beforeSendHooks.push(config.beforeSend);
+  }
+  if (config.ignoreErrors) {
+    beforeSendHooks.push(createBeforeSendHookFromIgnorePatterns(config.ignoreErrors));
+  }
 
   const add: Transports['add'] = (...transports) => {
-    value.push(...transports);
+    transports.push(...transports);
   };
 
-  const execute: Transports['execute'] = (payload) => {
-    value.forEach((transport) => transport.send(payload));
+  const execute: Transports['execute'] = (item) => {
+    let modified = beforeSendHooks.reduce<TransportItem | null>((prev, hook) => (prev ? hook(prev) : null), item);
```

Comment:
Should we switch from `reduce` to `for` iteration just to be able to exit faster from the loop?

Reply:
Definitely doens't matter for performance, unlikely to have more than 1 or 2 items. Is current version hard to read though?

Reply:
refactored. Feels like golang, verbose yet simple :thinking:  not bad